internal/uvm/create.go

```diff
@@ -208,17 +208,29 @@ func (uvm *UtilityVM) create(ctx context.Context, doc interface{}) error {
 }
 
 // Close terminates and releases resources associated with the utility VM.
-func (uvm *UtilityVM) Close() (err error) {
-	ctx, span := oc.StartSpan(context.Background(), "uvm::Close")
+func (uvm *UtilityVM) Close() error { return uvm.CloseCtx(context.Background()) }
+
+// CloseCtx is similar to [UtilityVM.Close], but accepts a context.
+//
+// The context is used for all operations, including waits, so timeouts/cancellations may prevent
+// proper uVM cleanup.
+func (uvm *UtilityVM) CloseCtx(ctx context.Context) (err error) {
+	ctx, span := oc.StartSpan(ctx, "uvm::Close")
 	defer span.End()
 	defer func() { oc.SetSpanStatus(span, err) }()
 	span.AddAttributes(trace.StringAttribute(logfields.UVMID, uvm.id))
 
+	// TODO: check if uVM already closed
+
 	windows.Close(uvm.vmmemProcess)
 
 	if uvm.hcsSystem != nil {
 		_ = uvm.hcsSystem.Terminate(ctx)
-		_ = uvm.Wait()
+		// uvm.Wait() waits on <-uvm.outputProcessingDone, which may not be closed until below
+		// (for a Create -> Stop without a Start), or uvm.outputHandler may be blocked on IO and
+		// take a while to close.
+		// In either case, we want to wait on the system closing, not IO completion.
+		_ = uvm.hcsSystem.WaitCtx(ctx)
```

Comment:
Wouldn't this block indefinitely too if the hcsSystem was created but never started?

Reply:
Nevermind. Turns out hcsSystem.Terminate will close the wait channel even if the system isn't started.